scanpointgenerator/excluders/roiexcluder.py

```diff
@@ -0,0 +1,65 @@
+from scanpointgenerator.core import Excluder, ROI
+from scanpointgenerator.compat import np
+
+
+@Excluder.register_subclass("scanpointgenerator:excluder/ROIExcluder:1.0")
+class ROIExcluder(Excluder):
+
+    """A class to exclude points outside of regions of interest."""
+
+    def __init__(self, rois, axes):
+        """
+        Args:
+            rois(list(ROI)): List of regions of interest
+            axes(list(str)): Names of axes to exclude points from
+
+        """
+        super(ROIExcluder, self).__init__(axes)
+
+        self.rois = rois
+
+    def create_mask(self, x_points, y_points):
+        """Create a boolean array specifying the points to exclude.
+
+        Args:
+            x_points(list(float)): List of points for x-axis
+            y_points(list(float)): List of points for y-axis
+
+        Returns:
+            np.array: Boolean array of points to exclude
+
+        """
+        if len(x_points) != len(y_points):
+            raise ValueError("Points lengths must be equal")
+
+        mask = np.zeros_like(x_points, dtype=bool)
+        for roi in self.rois:
+            # Accumulate all True entries
+            # Points outside of all ROIs will be excluded
+            mask |= roi.mask_points([x_points, y_points])
```

Comment:
The fact the excluder considers the union, not intersection, of the regions needs to be included in the docstrings.